core/sql/executor/ExStats.cpp

```diff
@@ -10666,6 +10688,27 @@ NABoolean ExMasterStats::filterForCpuStats(short subReqType,
       }
    }
    else
+   if (subReqType == SQLCLI_STATS_REQ_ACTIVE_QUERIES)
+   {
+      if (stmtState_ != Statement::PROCESS_ENDED_)
+      {
+         if (exeStartTime_ != -1) {
+            if (exeEndTime_ != -1)
+               tsToCompare = exeEndTime_;
+            else
+               tsToCompare = exeStartTime_;
+            lastActivity_ = (Int32)((currTimestamp-tsToCompare) / (Int64)1000000);
+            if (exeEndTime_ == -1)
+               return TRUE;
+            else
+            if (lastActivity_ <= etTimeInSecs) {
```

Comment:
So active queries with a default of 30 means all currently active queries (regardless of how long they have been active for) + queries that completed in last 30 seconds. Please ignore my previous comment.